src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -3,22 +3,62 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
 
+import java.util.Optional;
+
 import seedu.address.model.Model;
+import seedu.address.model.attribute.Attribute;
+
 
 /**
- * Lists all persons in the address book to the user.
+ * Lists all persons in the address book to the user. Attribute can be specified to show only specific attributes.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
 
-    public static final String MESSAGE_SUCCESS = "Listed all persons";
+    public static final String MESSAGE_SUCCESS = "Listed all clients";
+
+    public static final String MESSAGE_SUCCESS_ATTRIBUTE = "Listed all clients and filtered specified attribute";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Lists clients, along with specified attributes\n"
+            + "Parameters: -ATTRIBUTE (must be policy, phone or email)\n"
+            + "Example: " + COMMAND_WORD + " -policy";
 
+    private final Optional<Attribute> attribute;
+
+    /**
+     * @param attribute attribute that list will show
+     */
+    public ListCommand(Attribute attribute) {
+        requireNonNull(attribute);
+        this.attribute = Optional.of(attribute);
+    }
+
+    public ListCommand() {
+        this.attribute = Optional.empty();
+    }
+
+    public boolean isAttributeSpecified() {
+        return this.attribute.isPresent();
+    }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (this.attribute.isEmpty()) {
```

Comment:
A suggestion will be to do "if (!isAttributeSpecified()) {".

Reply:
Great suggestion, made the change.